models/branches.go

```diff
@@ -31,17 +31,18 @@ type ProtectedBranch struct {
 	BranchName                string `xorm:"UNIQUE(s)"`
 	CanPush                   bool   `xorm:"NOT NULL DEFAULT false"`
 	EnableWhitelist           bool
-	WhitelistUserIDs          []int64            `xorm:"JSON TEXT"`
-	WhitelistTeamIDs          []int64            `xorm:"JSON TEXT"`
-	EnableMergeWhitelist      bool               `xorm:"NOT NULL DEFAULT false"`
-	WhitelistDeployKeys       bool               `xorm:"NOT NULL DEFAULT false"`
-	MergeWhitelistUserIDs     []int64            `xorm:"JSON TEXT"`
-	MergeWhitelistTeamIDs     []int64            `xorm:"JSON TEXT"`
-	EnableStatusCheck         bool               `xorm:"NOT NULL DEFAULT false"`
-	StatusCheckContexts       []string           `xorm:"JSON TEXT"`
-	ApprovalsWhitelistUserIDs []int64            `xorm:"JSON TEXT"`
-	ApprovalsWhitelistTeamIDs []int64            `xorm:"JSON TEXT"`
-	RequiredApprovals         int64              `xorm:"NOT NULL DEFAULT 0"`
+	WhitelistUserIDs          []int64  `xorm:"JSON TEXT"`
+	WhitelistTeamIDs          []int64  `xorm:"JSON TEXT"`
+	EnableMergeWhitelist      bool     `xorm:"NOT NULL DEFAULT false"`
+	WhitelistDeployKeys       bool     `xorm:"NOT NULL DEFAULT false"`
+	MergeWhitelistUserIDs     []int64  `xorm:"JSON TEXT"`
+	MergeWhitelistTeamIDs     []int64  `xorm:"JSON TEXT"`
+	EnableStatusCheck         bool     `xorm:"NOT NULL DEFAULT false"`
+	StatusCheckContexts       []string `xorm:"JSON TEXT"`
+	ApprovalsWhitelistUserIDs []int64  `xorm:"JSON TEXT"`
+	ApprovalsWhitelistTeamIDs []int64  `xorm:"JSON TEXT"`
+	RequiredApprovals         int64    `xorm:"NOT NULL DEFAULT 0"`
+	GrantedApprovalsCount     int64
```

Comment:
This should not be added column so database so should be marked so:
```suggestion
	GrantedApprovalsCount     int64    `xorm:"-"`
```

Reply:
granted aprovals Should be stored for each PR and not for each branch?!

Reply:
And the information Iis stored already somewhere ... have to check where ...

Reply:
Aah, so any properties I add are mapped to the database _by default_?

Reply:
you can get the information via `models.GetUniqueApprovalsByPullRequestID(prID int64) ([]*Review, error)`

Reply:
and for the template engine you should provide them via `routers/repo/issue.go:`
as `IsBlockedByApprovals` value

Reply:
summary: 
* no changes in models/branches.go
* call models.GetUniqueApprovalsByPullRequestID(prID int64) in routers/repo/issue.go and store in `ctx.Data["GrantedApprovalsCount"]`

Reply:
> call models.GetUniqueApprovalsByPullRequestID(prID int64) in routers/repo/issue.go and store in ctx.Data["GrantedApprovalsCount"]

How would I reference that per PR, seeing as I have an array of them? Issues get popped under `ctx.Data["Issues"]` as a whole so I created a property to be able to access it from each one within the template. I presume something dynamic like `ctx.Data["Issues"][i].GrantedApprovals = count` isn't possible - I come from a JS background 😉.

The reason I left it as a property of `ProtectedBranch` was simply because `GetGrantedApprovalsCount` lived there already, and I just assumed I would keep them together.

Reply:
no

Reply:
I can look into it later ...

Reply:
I'm attempting to change it to a solution similar to `ctx.Data["CommitStatus"]` and removing the extra property on `ProtectedBranch`.

Looking at it, `GetUniqueApprovalsByPullRequestID` will return all approved reviews for a given PR, not just those on the approval whitelist. This might be equally valid - what would the user reasonably expect to see? Personally I think only reviews "which matter" should count to quickly determine if a PR is ready to be merged.